src/overlays/actors/ovl_En_Reeba/z_en_reeba.c

```diff
@@ -21,45 +40,617 @@ const ActorInit En_Reeba_InitVars = {
+                        if (this->unk_27E != 4) {
+                            this->unk_27E = 4;
+                            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+                            func_8003426C(&this->actor, 0, 0xFF, 0, 0x50);
+                            this->actionfunc = func_80AE58EC;
+                            break;
+                        }
+                    }
+                case 13: // hookshot/longshot
+                    if (this->actor.colChkInfo.health > 2) {
+                        if (this->unk_27E != 4) {
+                            this->unk_27E = 4;
+                            func_8003426C(&this->actor, 0, 0xFF, 0, 0x50);
+                            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+                            this->actionfunc = func_80AE58EC;
+                            break;
+                        }
+                    }
+                case 14:
+                    this->unk_27C = 6;
+                    Actor_ApplyDamage(&this->actor);
+                    if (this->actor.colChkInfo.health == 0) {
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_DEAD);
+                        func_80032C7C(globalCtx, &this->actor);
+                        this->actionfunc = func_80AE5BC4;
+                        break;
+                    }
+                    if (this->actionfunc == func_80AE5E48) {
+                        this->actor.shape.rot.z = 0;
+                        this->actor.shape.rot.x = this->actor.shape.rot.z;
+                    }
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_RIVA_DAMAGE);
+                    this->actionfunc = func_80AE57F0;
+                    break;
+                case 3: // ice arrows/ice magic
+                    Actor_ApplyDamage(&this->actor);
+                    this->unk_27C = 2;
+                    this->unk_27E = 2;
+                    func_8003426C(&this->actor, 0, 0xFF, 0, 0x50);
+                    this->actionfunc = func_80AE58EC;
+                    break;
+                case 1: // unknown
+                    if (this->unk_27E != 4) {
+                        this->unk_27E = 4;
+                        func_8003426C(&this->actor, 0, 0xFF, 0, 0x50);
+                        this->actionfunc = func_80AE58EC;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+void EnReeba_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnReeba* this = THIS;
+    Player* player = PLAYER;
+
+    func_80AE5EDC(this, globalCtx);
+    this->actionfunc(this, globalCtx2);
+    Actor_SetScale(&this->actor, this->scale);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Reeba/func_80AE4F40.s")
+    if (this->unk_270 != 0) {
+        this->unk_270--;
+    }
```

Comment:
Could these be DECRs ? 
```suggestion
    DECR(this->unk_270);
```

Reply:
they could but we never decided on using DECRs for stuff that isnt for an inline decrement in a condition. itd be nice if we did tho

Reply:
I think we should never use it for this because we can't do it consistently since it requires the check to be done as `!= 0` and that's not the only way to do it.

Reply:
we need to remove a few standalone DECRS in the codebase then. but yea ill revert it back

Reply:
I'm also not a fan of DECR for the != 0 vs > 0 issue but something to consider is that it really helps making code more redable imo, eg this DECR forest in `EnNiw_Update` https://github.com/Zelllll/oot/blob/en_niw/src/overlays/actors/ovl_En_Niw/z_en_niw.c#L941

<details>
<summary>click to show forest</summary>

```c
    EnNiw_ParticleUpdate(&this->actor, globalCtx);
    DECR(this->timer1);
    DECR(this->timer2);
    DECR(this->timer3);
    DECR(this->timer4);
    DECR(this->timer5);
    DECR(this->timer7);
    DECR(this->timer6);
    DECR(this->sfxTimer1);
    DECR(this->sfxTimer2);
    DECR(this->sfxTimer3);
    DECR(this->timer8);
    DECR(this->timer9);
    this->actor.shape.rot = this->actor.posRot.rot;
```

</details>